src/ray/object_manager/common.cc

```diff
@@ -180,17 +198,25 @@ Status PlasmaObjectHeader::ReadAcquire(
 
   // TODO(jhumphri): Wouldn't a futex be better here than polling?
   // Wait for the requested version (or a more recent one) to be sealed.
-  while (version < version_to_read || !is_sealed) {
-    RAY_CHECK_EQ(sem_post(sem.header_sem), 0);
-    sched_yield();
```

Comment:
this is something that should change looks like there's an old todo and was just a short-term hacky fix https://github.com/ray-project/ray/pull/44720, will look at a better way to do this, I think most times the scheduler will just try to run this again